### ISSUE TITLE
Latest version of pe_versions.json not being fetched from Forge API

### ISSUE THREAD

**Issue by sanfrancrisko** [puppetlabs/pdk#889]
**Describe the bug**
It appears as though the latest version of `pe_versions.json` is not being fetched from the [Forge API](https://forgeapi.puppet.com/private/versions/pe), which results in an error when trying to validate a module against a version of PE `> 2019.5`.

**To Reproduce**
- Install `pdk` version `1.18.0`
- Run `pdk validate --pe-version=2019.8.0` from a given module that is PDK compatible (e.g. `puppetlabs-apache`)

Observe the error:
```
pdk (ERROR): Unable to map Puppet Enterprise version 2019.8.0 to a Puppet version.
```

**Expected behaviour**
The `pe_versions.json` should be retrieved from the [Forge API](https://forgeapi.puppet.com/private/versions/pe) and pdk validate --pe-version=2019.8.0` should successfully execute against a given module

**Additional context**
- Running on `macOS 10.15.5` with `pdk 1.18.0` installed via `brew`
- Reinstallation of `pdk` did not resolve issue. Must ship with specific version of `pe_versions.json` and never get updated?
- Workaround by manually overwriting the `pe_versions.json` under PDK install root:
```shell
curl -k https://forgeapi.puppet.com/private/versions/pe -o /opt/puppetlabs/pdk/share/cache/pe_versions.json
```
- Initially reported via customer - see internal Slack thread [here](https://puppet.slack.com/archives/CF466M0F4/p1594285954302400)

**Comment by sanfrancrisko** [comment]
It appears as though we exit and return what's on disk for `pe_versions.json` [here](https://github.com/puppetlabs/pdk/blob/3d6a31093768d54a8dd86ed6a1a6f5f68c5f8c84/lib/pdk/util/vendored_file.rb#L20), because `PDK::Util.package_install?` returns `true`.

Is the expected behaviour that when the installation indicates it's a package install we should never fetch newer data remotely @scotje ? From reading [SDK-306](https://tickets.puppetlabs.com/browse/SDK-306) and reviewing #145 , it seems like we should have cached versions to make sure the PDK can operate with no external dependencies but it shouldn't prevent it updating if it can?

**Comment by scotje** [comment]
As you imply, this was somewhat intentional as we have users that use PDK in an air-gapped environment where there is no access to the public internet. At the time there wasn't an obvious place to introduce an update mechanism that wouldn't also incur a pause while we waited for a connection timeout for those users.

Perhaps we could introduce a config setting or other flag to allow users to update the cached file on demand or perhaps a setting to always try to look it up from Forge?

**Comment by DavidS** [comment]
It is very likely that the PDK doesn't contain the support bits for a specific pe/puppet version that has been released after its own release. Since we also tie commercial support for this, I'm reluctant to make it too easy to run an unsupported configuration.

The specific problem for 2019.8.0 has been addressed in the current nightlies, which will be released as PDK 2.0 soon:
```
david@zion:/tmp/test$ pdk validate --pe-version=2019.8.0
pdk (INFO): Puppet Enterprise 2019.8.0 maps to Puppet 6.19.1.
pdk (INFO): Using Ruby 2.7.2
pdk (INFO): Using Puppet 6.19.1
pdk (INFO): Running all available validators...
```
